languages/lang6.md

```diff
@@ -0,0 +1,35 @@
+## L6 Language
+
+```grammar
+.extends lang5
+```
+
+It's now possible to define global *variables*, which are locations that can be
+assigned to, read from, and have their address taken.
+
+Non-variables continue to works as they did previously.
+
+```grammar
+globalInit -= <dataInit>
+globaldef += (GlobalLoc <type> <dataInit>)
+
+dataInit -= (Data <int> <int>)
+          | (Data <int> (StringVal <string>))
+dataInit += (Data <type>)
+          | (Data <type> (StringVal <string>))
+```
+
+> TODO: instead of there being a `GlobalLoc`, `GlobalDef` should change its
+>       meaning to be that `GlobalLoc` from the L6 onwards. However, this is
+>       currently not possible, as some IL producers (mis-)use the globals for
+>       passing constants with external meaning to the VM
```

Comment:
Would this imply some sort of extern facility, where we can mark certain locations as need to be kept stable for the host?

Reply:
It wouldn't, no.

Exportable constants are only needed for the host memory configuration variables, two of which (i.e., `total_memory` and `stack_start`) are now obsolete. The stack size configuration variable (`stack_size`) will be replaced with some dedicated language/module feature in the future, but I haven't fully figured out the details yet.

In other words, exportable constants in their current form are moribund. Once they're all gone, `GlobaLoc` can be removed again, with `GlobalDef` taking over its meaning.